src/placeos-pulse/client.cr

```diff
@@ -32,12 +33,14 @@ module PlaceOS::Pulse
     def initialize(
       @instance_token : String,
       @email : String,
+      private_key : String,
```

Comment:
```suggestion
      private_key : Sodium::Sign::SecretKey,
```

I think keeping the key in the object form may save some unnecessary object construction.